libraries/Microsoft.Bot.Builder.Dialogs/DialogExtensions.cs

```diff
@@ -79,8 +82,29 @@ internal static async Task<DialogTurnResult> InternalRunAsync(ITurnContext turnC
                     var innerExceptions = new List<Exception>();
                     try
                     {
+                        errorHandlerCalled++;
+
                         // fire error event, bubbling from the leaf.
                         handled = await dialogContext.EmitEventAsync(DialogEvents.Error, err, bubble: true, fromLeaf: true, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                        var turnObject = (JObject)turnContext.TurnState["turn"];
```

Comment:
cant we use `context.TurnState.Get<bool>(TurnPath.ExecutionLimit)` ? to get the executionLimit?

Reply:
I tried something similar at the beginning, but the Get<T> is expecting a class. It doesn't work with <int>.